sqs/README.md

```diff
@@ -2,36 +2,237 @@
+
+The producer file would look like this:
+
+```yaml
+channels:
+  user-signedup:
+    bindings:
+      sqs:
+        queue:
+          name: user-signedup-queue
+          type: standard
+          receiveMessageWaitTime: 4
+          reDrivePolicy:
+            deadLetterQueue:
+              name: user-signedup-dlq
+          policy:
+            statements: 
+            - effect : Allow
+              principal: *
+              action: Sqs:SendMessage
+            - effect : Allow
+              principal: *
+              action: Sqs:ReceiveMessage
+        deadLetterQueue: 
+          name: user-signedup-dlq 
+          messageRetentionPeriod: 1209600 
+	        type: standard
+    subscribe:
+      operationId: sendMessage
+      description: sends messages when a user has signed up
+      bindings:
+        sqs: {}
+
+```
+In this case we can minimize duplicated information by omitting the binding in our specification, and assume it is picked up from the producer file. We can use an empty object to indicate the SQS Binding on the **publish** Operation Object, if need a marker for generation, otherwise we could omit the Operation Binding Object.
+
+```yaml
+channels:
+  user-signedup:
+   publish:
+      operationId: receiveMessage
+      description: receives a messages when a user has signed up
+      bindings:
+        sqs: {}
+
+```
+
+#### SNS to SQS Pub-Sub
+
+[<img src="SNS-SQS-Pub-Sub.png" height = "400" width="600"/>](SNS-SQS-Pub-Sub.png)
+
+In this example, we are using SNS for the channel, and SQS to receive from SNS. 
+
+
+The producer files looks like this (see the [SNS Binding]() for more).
+
+```yaml
+channels:
+  user-signedup:
+    description:  A user has signed up for our service
+    binding :
+      sns: {}     # Indicates that the channel is an SNS Topic
+    subscribe:
+      operationId: sendMessage
+      description: send messages to the topic
+      bindings:
+        sns:
+          policy:
+            statements: 
+            - effect : Allow
+              principal: *
+              action: SNS:Publish
+```
+
+And the consumer file would look like this. Note that for simplicity, we choose not to repeat the SNS Binding on the Consumer as it does not 'own' the channel.
+
+
+```yaml
+channels:
+  user-signedup:
+    description:  A user has signed up for our service
+    publish:
+      operationId: receiveMessage
+      description: receive messages from the topic
+      bindings:
+        sns:
+          consumers:
+          - protocol: sqs
+            endpoint:
+            name: user-signedup-queue
+            rawMessageDelivery: true  
+            filterPolicy:
+              attributes:
+                reason: 
+                  anything-but: password-reset
+            reDrivePolicy:
+              deadLetterQueue:
+              name: user-signedup-queue-dlq
+        sqs:
+          queues:
```

Comment:
Hope my comment above addressed this. At the `publish` level you can configure multiple queues that receive data, but at the channel level you should only configure a single queue because this is the primary logical pipe through which data flows. For the example a point to point SQS model (in the single channel level queue example) as opposed to SNS -> SQS pub sub model (possible one to many messaging).